File: hot-framework/src/test/resources/be/icode/hot/test/nio/http/py/http-client-post-form.py
print("Starting...")

options = {
	'url': 'http://localhost:8080/test',
	'type': 'POST',
	'headers':{
		'Content-type':'application/x-www-form-urlencoded'
	},
	'data':{
		'name':'Damien',
		'age':8
	}
}

def done(data, status, response):
    print data
    server.stop()

def fail(response, status, error):
    print "error"
    server.stop()

client.buildRequest(options).done(done).fail(fail)

File: hot-framework/src/test/resources/be/solidx/hot/test/nio/http/js/http-client-post-form.js
hprint("Starting...")

var options = {
	url: 'http://localhost:8080/test',
	type: 'POST',
	headers:{
		'Content-type':"application/x-www-form-urlencoded"
	},
	data:{
		name:'Damien',
		age:8
	}
}

client.buildRequest(options).done(function(data) {
	hprint(new java.lang.String(data))
	server.stop()
}).fail(function(response, status, error) {
	hprint(error)
	server.stop()
}) 